x-pack/plugins/siem/public/timelines/components/timeline/body/events/event_column_view.tsx

```diff
@@ -79,17 +86,70 @@ export const EventColumnView = React.memo<Props>(
     toggleShowNotes,
     updateNote,
   }) => {
-    const timelineTypeContext = useTimelineTypeContext();
-
+    const { getManageTimelineById } = useManageTimeline();
+    const timelineActions = useMemo(() => getManageTimelineById(timelineId).timelineRowActions, [
+      getManageTimelineById,
+      timelineId,
+    ]);
     const additionalActions = useMemo<JSX.Element[]>(() => {
-      return (
-        timelineTypeContext.timelineActions?.map((action) => (
-          <EventsTdContent key={action.id} textAlign="center">
-            {action.getAction({ eventId: id, ecsData })}
-          </EventsTdContent>
-        )) ?? []
+      const grouped = timelineActions.reduce(
+        (
+          acc: {
+            contextMenu: JSX.Element[];
+            icon: JSX.Element[];
+          },
+          action
+        ) => {
+          if (action.displayType === 'icon') {
+            return {
+              ...acc,
+              icon: [
+                ...acc.icon,
+                <EventsTdContent key={action.id} textAlign="center">
+                  <EuiToolTip
+                    data-test-subj={`${action.dataTestSubj}-tool-tip`}
+                    content={action.content}
+                  >
+                    <EuiButtonIcon
+                      aria-label={action.ariaLabel}
+                      data-test-subj={`${action.dataTestSubj}-button`}
+                      iconType={action.iconType}
+                      isDisabled={action.isActionDisabled ?? false}
+                      onClick={() => action.onClick({ eventId: id, ecsData })}
+                    />
```

Comment:
Seeing the ole `Warning: Each child in a list should have a unique "key" prop.` error in the console when first opening the menu -- should be fine to just add `key={action.id}` as that should be unique, correct?